### PR TITLE
01 clean split data

### PR DESCRIPTION
**Pull request by bill-amadio**
#### Purpose/implementation

_What scientific question is your analysis addressing? 
What was your approach?
If this is not adding an analysis, describe your changes in this section._

This is a script to read a gene expression file and drop samples from it that have invalid entries in either the reported_gender or germline_sex estimate column of the histologies file.

The remaining samples in the gene expression file are split into train and test sets according to a user-specified partition scheme.

Finally, the training set, the test set and a TSV file with reported_gender and germline_sex_estimate values are saved in a user-specified output directory.

#### Issue

_What GitHub issue does your pull request address?_

#84 RNA-Expression based Prediction of Sex

#### Directions for reviewers

_Tell potential reviewers what kind of feedback you are soliciting. 
Are there particular areas that need a closer look?
Is there something you want to discuss further?
Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?_

Accuracy, efficiency and generalizability

#### Results

_If your pull request includes code that produces scientific results, please summarize the results here.
This can help facilitate discussion around interpretation.
Please state what kinds of results are included (e.g., table, figure)._

#### Docker and continuous integration

_Check all those that apply or remove this section if it is not applicable._

- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [x] This analysis has been added to continuous integration.

### CODE REVIEW

File: analyses/sex-prediction-from-RNASeq/01-clean_split_data.R [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+  test_set <- input_mat[test_set_index, ]
+} else {
+  train_set <- input_mat
+}
+#--------
+
+#--------Save train, (test, if used) and targets files to output directory
+
+if (train_percent < 1) {
+  save(train_set, test_set, file=paste("../../", output_directory, "/", "train_test_data.RData", sep=""))
```
[the remark this answers is not in the document]
Comment (jaclyn-taroni) [reply analyses/sex-prediction-from-RNASeq/01-clean_split_data.R]: Could also use `write_rds` from `readr` package here if you'd prefer.

File: analyses/sex-prediction-from-RNASeq/01-clean_split_data.R [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+} else {
+  save(train_set, file=paste("../../", output_directory, "/", "train_data.RData", sep=""))
+}
+
+write_tsv(targets, paste("../../", output_directory, "/", "targets.tsv", sep=""),
```
Comment (jaclyn-taroni) [inline analyses/sex-prediction-from-RNASeq/01-clean_split_data.R]: I would probably change this line to use `filename_lead` as well.

**Comment by bill-amadio** [comment]
All suggested changes above have been made.  Script now runs with command line arguments.  I pasted Jackie's code and made changes as needed downstream in the script.

File is attached with .txt extension to allow attaching.  Changes have been pushed to 
OpenPBTA-analysis/analyses/sex-prediction-from-RNASeq/01-clean_split_data.R 

[01-clean_split_data.R.txt](https://github.com/AlexsLemonade/OpenPBTA-analysis/files/3764116/01-clean_split_data.R.txt)

**Comment by jaclyn-taroni** [comment]
I am going to update this branch and add this script to CI.

**Comment by bill-amadio** [comment]
Thank you.  I'll can move on to 02 next week.

File: analyses/sex-prediction-from-RNASeq/01-clean_split_data.R [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+
+targets_unsorted <- histologies[histologies$Kids_First_Biospecimen_ID %in% rownames(input_mat), 
+                               c("Kids_First_Biospecimen_ID", "reported_gender", "germline_sex_estimate")]
+
+#Check sequence of rownames(input_mat) and targets[, 1].  
+#reported_gender_response holds the reported_gender values in the same Kids_First_Biospecimen_ID sequence as rownames(df).
+
+match_index <- unlist(sapply(rownames(input_mat), function(x) which(targets_unsorted[, 1] == x)))
+
+targets <- targets_unsorted[match_index, c(1:ncol(targets_unsorted))]
```
[the remark this answers is not in the document]
Comment (jaclyn-taroni) [reply analyses/sex-prediction-from-RNASeq/01-clean_split_data.R]: This sounds good to me @bill-amadio! Do note that I added your script to CI, so I made commits to your branch that you may want to pull before making those changes.

[the remark this answers is not in the document]
Comment (bill-amadio) [reply analyses/sex-prediction-from-RNASeq/01-clean_split_data.R]: Added code for sequencing expression and target files.  Pushed changes.
